h5grove/models.py

```diff
@@ -40,3 +41,54 @@ class LinkResolution(str, Enum):
     },
     total=False,
 )
+
+
+class EntityMetadata(TypedDict):
+    name: str
+    kind: str
+
+
+class ExternalLinkMetadata(EntityMetadata):
+    target_file: str
+    target_path: str
+
+
+class SoftLinkMetadata(EntityMetadata):
+    target_path: str
+
+
+AttributeMetadata = TypedDict(
```

Comment:
Is there a problem if you use the `class` syntax here ?

Reply:
Nope, just thought the idea was to use he short syntax when possible ... but reading https://typing.readthedocs.io/en/latest/spec/typeddict.html#typeddict this doesn't seem to be the case. Should I change it?

Reply:
I think it is best to be consistent and use the `class` syntax everywhere. This alternative syntax is apparently for Python version that are even older than 3.8.

Reply:
+1

Reply:
Actually, can't use the `class` syntax for `TypeMetadata` because of the `class` key - cf. https://docs.python.org/3/library/typing.html#typing.TypedDict

Reply:
Ah yes of course. Good enough :+1: